pkg/hwinfo/hwinfo_windows.go

```diff
@@ -0,0 +1,157 @@
+// +build windows
+
+package hwinfo
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/StackExchange/wmi"
+)
+
+type winMemoryType uint16
+
+type win32_PhysicalMemory struct {
+	BankLabel     string
+	Capacity      uint64
+	DataWidth     uint16
+	DeviceLocator string
+	InstallDate   *time.Time
+	Manufacturer  *string
+	MaxVoltage    *uint32
+	MinVoltage    *uint32
+	MemoryType    winMemoryType
+	Model         *string
+	PartNumber    *string
+	SerialNumber  *string
+	Speed         uint32
+	Status        *string
+	TotalWidth    *uint16
+}
+
+type win32_BaseBoard struct {
+	Manufacturer string
+	Model        *string
+	SerialNumber string
+}
+
+type win32_Processor struct {
+	DeviceID                  *string
+	Description               string
+	Name                      string
+	Manufacturer              string
+	SerialNumber              string
+	SocketDesignation         string
+	MaxClockSpeed             uint32
+	NumberOfCores             uint32
+	NumberOfEnabledCore       uint32
+	NumberOfLogicalProcessors uint32
+	ProcessorType             uint16
+}
+
+func (w winMemoryType) String() string {
+	switch w {
+	case 2:
+		return "DRAM"
+	case 3:
+		return "Synchronous DRAM"
+	case 4:
+		return "Cache DRAM"
+	case 5:
+		return "EDO"
+	case 6:
+		return "EDRAM"
+	case 7:
+		return "VRAM"
+	case 8:
+		return "SRAM"
+	case 9:
+		return "RAM"
+	case 10:
+		return "ROM"
+	case 11:
+		return "FLASH"
+	case 12:
+		return "EEPROM"
+	case 13:
+		return "FEPROM"
+	case 14:
+		return "EPROM"
+	case 15:
+		return "CDRAM"
+	case 16:
+		return "3DRAM"
+	case 17:
+		return "SDRAM"
+	case 18:
+		return "SGRAM"
+	case 19:
+		return "RDRAM"
+	case 20:
+		return "DDR"
+	case 21:
+		return "DDR2"
+	case 22:
+		return "DDR2 FB-DIMM"
+	case 24:
+		return "DDR3"
+	case 25:
+		return "FBD2"
+	default:
+		return "unknown"
+	}
+}
+
+func fetchInventory() (map[string]interface{}, error) {
+	res := make(map[string]interface{})
+
+	var cpus []win32_Processor
+	query := wmi.CreateQuery(&cpus, "")
+	err := wmi.Query(query, &cpus)
```

Comment:
We used the `WMIQueryWithContext` in other parts of the code. This is Windows. Nobody knows for sure how many years the WMI query could take :)
https://github.com/cloudradar-monitoring/cagent/blob/master/processes_windows.go#L23

Maybe we can move this function to `pkg/wmi/wmiutil`

Reply:
Ok